backend/app/api/api_v1/endpoints/users.py

```diff
@@ -0,0 +1,160 @@
+            if is_user_active(user):
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST, detail=f"User {github_id} is already active"
+                )
+            user.is_active = True
+            sess.add(user)
+            sess.commit()
+            return User.model_validate(user)
+    except ValidationError:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+
+
+@router.patch("/{github_id}/deactivate", response_model=User, description="Deactivate a user")
+async def deactivate_user(github_id: int) -> User:
+    try:
+        with get_sess() as sess:
+            user = sess.query(mUser).filter(mUser.github_id == github_id).first()
+            if not user:
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+            if not is_user_active(user):
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST, detail=f"User {github_id} is already inactive"
+                )
+            user.is_active = False
+            sess.add(user)
+            sess.commit()
+            return User.model_validate(user)
+    except ValidationError:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+
+
+@router.patch("/{github_id}/role", response_model=User, description="Set user a role")
+async def set_user_role(github_id: int, role: Role) -> User:
+    if role not in utils.get_roles():
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=f"Role {role} not found")
+    try:
+        with get_sess() as sess:
+            user = sess.query(mUser).filter(mUser.github_id == github_id).first()
+            if not user:
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+            user.role = role
+            sess.add(user)
+            sess.commit()
+            return User.model_validate(user)
+    except ValidationError:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+
+
+@router.patch("/{github_id}/", response_model=User, description="Update user data")
+async def update_user_data(github_id: int, payload: dict[str, Any]) -> User:
+    try:
+        user = utils.get_user(github_id)
+    except ValidationError:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"User {github_id} not found")
+    try:
+        for key, value in payload.items():
+            setattr(user, key, value)
+        return utils.update_user(user)
+    except ValidationError:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Bad request data")
+    except ValueError:
+        raise HTTPException(status_code=status.HTTP_417_EXPECTATION_FAILED, detail="Invalid data")
```

Comment:
It would be better for the user experience to display what went wrong so that users can easily identify their mistakes. This should be straightforward with Pydantic